src/microbe_stage/systems/MicrobeAISystem.cs

```diff
@@ -1320,34 +1419,45 @@ private void BuildChunksCache()
         // To allow multithreaded AI access safely
         lock (chunkDataCache)
         {
-            if (chunkCacheBuilt)
-                return;
-
-            foreach (ref readonly var chunk in chunksSet.GetEntities())
+            lock (terrainChunkDataCache)
             {
-                if (chunk.Has<TimedLife>())
+                if (chunkCacheBuilt)
+                    return;
+
+                foreach (ref readonly var chunk in chunksSet.GetEntities())
                 {
-                    // Ignore already despawning chunks
-                    ref var timed = ref chunk.Get<TimedLife>();
+                    if (chunk.Has<TimedLife>())
```

Comment:
Is this a new code? Is this really that useful to add here? This means that relatively the chunk cache will take quite a bit longer to build. And my hunch is that this doesn't help much as it takes just a few seconds for chunks to dissolve, so microbes don't see that many disappearing chunks even without this check.

Reply:
Yes, this code was already there, but I can remove it

Reply:
If the code was there already, then fine it can stay.
The diff should get a bit clearer once you remove that one lock and deindent a lot of the lines back to what they were.